benchmarking/data/simulated_data/mock_student_provider.py

```diff
@@ -55,27 +60,52 @@ def validate(self):
                 f"num_project_preferences_per_student ({self.num_project_preferences_per_student}) cannot "
                 "be > the number of project options."
             )
-        if not is_non_negative_integer(self.num_project_preferences_per_student):
-            raise ValueError(
-                f"num_project_preferences_per_student ({self.num_project_preferences_per_student}) must be a "
-                f"non-negative integer."
-            )
         if not is_unique(self.project_preference_options):
             raise ValueError(f"project_preference_options must be unique if specified.")
         if self.ensure_exact_attribute_ratios and self.num_values_per_attribute:
             raise ValueError(
                 "Cannot ensure exact attribute ratios when specifying num_values_per_attribute"
             )
 
+        if not is_unique(list(self.attribute_ranges.keys())):
```

Comment:
Wait but it's an object? Like a hashmap? If you set 2 keys as the same it would have overwritten it, no?